src/main/java/io/confluent/idesidecar/restapi/credentials/TLSConfig.java

```diff
@@ -0,0 +1,375 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import com.fasterxml.jackson.annotation.JsonEnumDefaultValue;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.confluent.idesidecar.restapi.exceptions.Failure;
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.soabase.recordbuilder.core.RecordBuilder;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Null;
+import jakarta.validation.constraints.Size;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+/**
+ * The SSL/TLS configuration object.
+ * Usage modes:
+ * <ul>
+ *   <li>Default SSL settings without truststore or keystore: set {@link #enabled} to {@code true}
+ *   and leave other fields unset</li>
+ *   <li>Truststore only: provide a truststore path and password</li>
+ *   <li>Truststore and keystore: provide a truststore path and password, and a keystore path and
+ *   password. This is equivalent to mutual TLS or mTLS</li>
+ *   <li>Disable hostname verification: set {@link #verifyHostname} to {@code false} to disable
+ *   server certificate hostname verification</li>
+ *   <li>SSL Disabled: set {@link #enabled} to {@code false} to disable SSL</li>
+ *</ul>
+ */
+@Schema(description = "SSL configuration")
+@RecordBuilder
+public record TLSConfig(
+    @Schema(
+        description = "Whether to verify the server certificate hostname."
+        + " Defaults to true if not set.",
+        defaultValue = DEFAULT_VERIFY_HOSTNAME_VALUE
+    )
+    @JsonProperty(value = "verify_hostname")
+    @Null
+    Boolean verifyHostname,
+
+    @Schema(
+        description = "Whether SSL is enabled. If not set, defaults to true.",
```

Comment:
The second sentence is contradictory to requiring this field to be set.
I'd suggest to drop the constraint `required = true` and change `@NotNull` to `@Null`.